### PR TITLE
feat(ABS): replace legacy azure-storage lib with @azure/storage-blob

### PR DESCRIPTION
**Pull request by konclave**
## In this PR:
- Legacy `azure-storage` library was replaced with `@azure/storage-blob`. 
- azure-blob-storage module tests were refactored. The previous implementation had the tests that were dependent on each other and failed if you changed the tests order. 

## Issues reference:
- [https://github.com/ducktors/turborepo-remote-cache/issues/186](https://github.com/ducktors/turborepo-remote-cache/issues/186)

### Checklist:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/ducktors/turborepo-remote-cache/pulls) for the same update/change?
* [x] Have you lint your code with `pnpm lint` locally prior to submission?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran build with `pnpm build` of your changes locally?
* [x] Have you successfully ran tests with `pnpm test` of your changes locally?
* [x] Have you commit using [Conventional Commits](https://github.com/ducktors/turborepo-remote-cache#how-to-commit)?

### CODE REVIEW

File: test/azure-blob-storage.ts [hunk 1]
```diff
@@ -1,10 +1,7 @@
 import assert from 'node:assert'
 import crypto from 'node:crypto'
-import fs from 'node:fs'
-import { tmpdir } from 'node:os'
-import { join } from 'node:path'
-import { mock, test } from 'node:test'
-import fsbs from 'fs-blob-store'
```
Comment (konclave) [inline test/azure-blob-storage.ts:7]: Testing with reading/writing the real file on the filesystem doesn't look reliable. One test pass depends on the other test run, tests fail if you change the order and using another lib in the azure storage mock adds unnecessary complexity. It's enough to check if the mock was called with the expected arguments. 
I suggest to refactor the other test suites the same way.

Comment (fox1t) [reply test/azure-blob-storage.ts:7]: This is a good point.

**Review by fox1t — APPROVED** [review]
LGTM, thanks for this contribution!

**Pull request merged**